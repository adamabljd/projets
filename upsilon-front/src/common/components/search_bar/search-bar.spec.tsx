import React from 'react';
import { fireEvent, render, RenderResult, screen } from '@testing-library/react';
import SearchBar, { SearchBarProps } from './search-bar.component';

const mockHandleOnSubmit = jest.fn();

const mockSearchBarProps: SearchBarProps = {
  placeHolder: 'place Holder',
  handleOnSubmit: mockHandleOnSubmit
};

function renderSearchBar(props: SearchBarProps): RenderResult {
  return render(<SearchBar {...props} />);
}

describe('SearchBar', () => {
  describe('static rendering', () => {
    it('should render Search Bar', () => {
      const { baseElement } = renderSearchBar(mockSearchBarProps);
      expect(baseElement).toMatchSnapshot();
    });
  });

  describe('SearchBarButtonSubmit', () => {
    it('should submit form click on submit button', () => {
      renderSearchBar(mockSearchBarProps);
      fireEvent.click(screen.getByRole('button'));
      expect(mockHandleOnSubmit).toHaveBeenCalled();
    });
  });
});
