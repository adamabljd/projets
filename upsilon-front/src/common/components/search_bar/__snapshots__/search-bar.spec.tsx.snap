// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar static rendering should render Search Bar 1`] = `
<body>
  <div>
    <form
      class="search-bar"
    >
      <input
        class="input-text"
        name="search-bar-input"
        placeholder="place Holder"
        type="text"
        value=""
      />
      <hr
        class="vertical-line"
      />
      <button
        class="search-button"
        type="submit"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-magnifying-glass search-icon"
          data-icon="magnifying-glass"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352a144 144 0 1 0 0-288 144 144 0 1 0 0 288z"
            fill="currentColor"
          />
        </svg>
      </button>
    </form>
  </div>
</body>
`;
